lib/main.js

```diff
@@ -27,6 +27,7 @@ function log(x) {
 const ERROR_NO_MATCHING_CERT = "error:noMatchingCert";
 const ERROR_USER_CANCEL = "error:userCancel";
 const ERROR_INTERNAL = "error:internalError";
+const ERROR_AUTHENTICATION_FAILED = "error:authenticationFailed";
```

Comment:
While there is no standard for this functionality, using this error code will deviate from the behavior described at https://docs.oracle.com/cd/E19957-01/816-6152-10/ (which is as close to an authoritative source as exists). I think that's probably fine, but we should keep an eye out for errors potentially caused by an unexpected error string.